base/docs/utils.jl

```diff
@@ -72,6 +72,9 @@ print(io::IO, t::Text) = print(io, t.content)
 print{F <: Function}(io::IO, t::Text{F}) = t.content(io)
 show(io::IO, t::Text) = print(io, t)
 
+=={T<:Union{HTML, Text}}(t1::T, t2::T) = t1.content == t2.content
+hash{T<:Union{HTML, Text}}(t::T, h::UInt) = hash(t.content, h)
```

Comment:
Probably want to mix in a static hash of the type here unless you want cross-type values to hash the same (e.g. `hash(HTML("foo")) == hash(Text("foo")) == hash("foo")`. As an aside, we really need to make this easier to do correctly.